### PR TITLE
Mem stats Implementation on OSX

### PR DESCRIPTION
**Pull request by apingault**
BEGINRELEASENOTES
- Fix compilation on OSX: `sysinfo.h` header doesn't exist on OSX  
- Implements memory stat on OSX  [WIP]

ENDRELEASENOTES

Some context on the apple way of managing memory, more information available on the apple developer [documentation](https://developer.apple.com/library/content/documentation/Performance/Conceptual/ManagingMemory/Articles/AboutMemory.html) :
 - `Swap`: It's allocated dynamically and increases as the system needs it until there is no more room on the boot disk : I still count the allocated swap in the total `Virtual Memory` allocated.
 - Total Memory:
   - `Wired Memory`: Physical Memory in use that cannot go to swap
   - `Active Memory`: Physical Memory in use that can go inactive if not accessed for some time (`App Memory` in Activity Monitor)
   - `Inactive Memory`: Physical Memory that might go to swap if needed (`Cached Memory` in the Activity Monitor)
   - `Free Memory`: Truly free ram that can be allocated to a process without needing to swap
   - On top of the `active/inactive/wired` memory there is the `Compressed Memory`, that is part of the Physical Memory (not in the swap) compressed to make more room for other processes. I didn't find a way to access its value that can actually be quite extensive (~2Gb currently on my 8Gb RAM mac for instance).  
- Process Memory:
   - `Physical Memory`: corresponds to the real memory foot print (as can be found in `real mem` of the Activity Monitor).
   - `Virtual Memory`: Does not have the same meaning as on unix. Every process seems to have around 4GB (on my mac) of `VM` allocated. I guess we could display the `Physical Memory` here too, so the readings make sense.

### CODE REVIEW

**Comment by rete** [comment]
By using `dqm4hep-memstats`, does the numbers make sense to you ?

File: source/src/Internal.cc [hunk 1]
```diff
@@ -62,9 +62,62 @@ namespace dqm4hep {
...
-      /* MACOSX implementation: TBD */
+    void memStats(MemoryStats &stats) {
+      double unit = 1024. * 1024.; // Store everything in mb
+
+      // Swap info
+      // Swap is dynamically alotted on OSX, It will expand as the system needs it until the boot drive is filled.
+      struct xsw_usage swapStats;
+      size_t lengthSwap = sizeof(swapStats);
+      if (KERN_SUCCESS != sysctlbyname("vm.swapusage", &swapStats, &lengthSwap, NULL, 0))
+        std::cerr << "[OSX] - Failed to get swap statistics" << std::endl;
```
Comment (rete) [inline source/src/Internal.cc]: Can you replace these `std::cerr` by `dqm_error("message")` and throw a `StatusCodeException` ? Same for every call below

Comment (apingault) [reply source/src/Internal.cc]: Yes, this was temporary and forgot to change it. Fixed in https://github.com/DQM4HEP/dqm4hep-core/pull/34/commits/22cf2aee7d2b138c40605e89cc81b406c796df6d

**Comment by apingault** [comment]
>By using `dqm4hep-memstats`, does the numbers make sense to you ?

Yes, the only grudge I have is the displayed virtual memory for the process. I was hoping to get the same information as on the OSX GUI (`Memory` column in the Activity Monitor) but I can't find a way to get this. 
Tools like`top` or `ps` doesn't report this information either (they have the same ~4Go virtual memory as I get) so I guess we can't access it.
Final choice : I display physical memory instead, this looks more meaning full.

**Review by rete — APPROVED** [review]

**Pull request merged**